src/retention/organization_retention_handler.py

```diff
@@ -33,6 +40,13 @@ def __schedule_retention_starting_from_cursor(cls, table_cursor):
 
             Tasks.schedule(queue_name='table-retention-scheduler', tasks=[task])
 
+    @staticmethod
+    def table_with_backup():
```

Comment:
I would move this method to Table class and change it name to "table_has_backups()"

Reply:
I divided handler and service part. I refactored method mentioned by you (rename, making private, and extract variable to making logic more understandable). I still think that this predicate method should stay in that class, as it is strongly connected with retention process optimization.